File: 101_200/121_Best_Time_to_Buy_and_Sell_Stock/Solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l,i,m,rm = len(prices), len(prices)-1, 0,0
        if l > 1:
            while i >= 0:
                m = max(0, max(m, rm - prices[i]))
                rm = max(rm, prices[i])
                i-=1
        return m

File: 101_200/121_Best_Time_to_Buy_and_Sell_Stock/Solution.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut max = 0i32;
        for i in 0..prices.len() {
            let mut left_max = 0i32;
            let pos = i.min(prices.len());
            for j in pos..prices.len() {
                let delta = prices[j] - prices[i];
                left_max = delta.max(0).max(left_max);
            }
            max = left_max.max(max);
        }
        max
    }
}
